### ISSUE TITLE
AttributeError on import in 8.0.14 when using PyTorch<1.6.0 (?)

### ISSUE THREAD

**Issue by lumi-tworthey** [explosion/thinc#615]
As of the v8.0.14, the following traceback is occurring at import time in a project that has PyTorch pinned to v1.4.0:

```
.venv/lib/python3.6/site-packages/thinc/util.py:34: in <module>
    has_torch_amp = not torch.cuda.amp.common.amp_definitely_not_available()
AttributeError: module 'torch.cuda' has no attribute 'amp'
```

It seems `torch.cuda.amp` did not exist in torch [until version 1.6.0](https://github.com/pytorch/pytorch/releases/tag/v1.6.0). The offending line (at `util.py:34`) is guarded by an `except ImportError`, but not `AttributeError` for the cases in which the installed torch version does not know about the module being accessed in the first place

```
    has_torch_amp = not torch.cuda.amp.common.amp_definitely_not_available()
```

**Comment by adrianeboyd** [comment]
Hi, we're working on a fix in #610. Hopefully we'll have a new release very soon. In the meanwhile v8.0.13 should still be compatible with older versions of pytorch.

**Comment by adrianeboyd** [comment]
Please try upgrading to v8.0.15, which should hopefully resolve the `AttributeError` on import.

However, after reviewing the changes and running more thorough tests of all recent torch releases, we've revised the lower recommended torch version to v1.6.0, so I'm not sure whether you still might run into other errors with v1.4.0.
